dgraphHelper.js

```diff
@@ -0,0 +1,370 @@
+import diff from 'deep-diff';
+import dgraph from 'dgraph-js';
+import grpc from 'grpc';
+
+import { SCHEMA, TYPES } from './schema';
+
+const DB_URL = process.env.DGRAPH_URL;
+
+const create_client = () => {
+  const CLIENT_STUB = new dgraph.DgraphClientStub(DB_URL, grpc.credentials.createInsecure());
+  return new dgraph.DgraphClient(CLIENT_STUB);
+}
+
+const get_schema = async client => (await client.newTxn().query('schema {}')).getJson();
+
+const prepare_value_string = predicate => {
+  const PREPARED_ARRAY = predicate.split(' ');
+  if (PREPARED_ARRAY[PREPARED_ARRAY.length - 1] === '.') {
+    PREPARED_ARRAY.pop();
+  } else {
+    throw new Error('Missing . at the end of predicate, or incorrect spacing.');
+  }
+  return PREPARED_ARRAY;
+};
+
+const type_check = (type, object) => {
+  const NORMAL_TYPES = ['default', 'bool', 'datetime', 'float', 'geo', 'int', 'password', 'string', 'uid'];
+  const LIST_TYPES = ['[default]', '[bool]', '[datetime]', '[float]', '[geo]', '[int]', '[string]', '[uid]'];
+  const TYPE_IS_NOT_ALIST = NORMAL_TYPES.some(value => value === type);
+  const TYPE_IS_LIST = LIST_TYPES.some(value => value === type);
+  if (!TYPE_IS_NOT_ALIST && !TYPE_IS_LIST) {
+    throw new Error('Incorrect or missing type in predicate.');
+  } else if (TYPE_IS_LIST) {
+    object.type = type.slice(1, -1);
+  } else {
+    object.type = type;
+  }
+  if (TYPE_IS_LIST) {
+    object.list = true;
+  }
+};
+
+const index_check = (aValues, object) => {
+  const TOKENIZER_ARRAY = [];
+  const INDEX = aValues.some(value => {
+    if (value.includes('@index')) {
+      if (value.slice(6, 7) !== '(' || value.slice(-1) !== ')') {
+        throw new Error('@index is invalid, missing parenthesis or there are spaces in tokenizer.');
+      }
+      const FIELDS = value.slice(7, -1).split(',');
+      FIELDS.forEach(field => {
+        TOKENIZER_ARRAY.push(field.trim());
+      });
+      return true;
+    }
+    return false;
+  });
+  if (INDEX) {
+    object.index = INDEX;
+    object.tokenizer = TOKENIZER_ARRAY;
+  }
+};
```

Comment:
You should not modify `object`

You can use map & flatMap instead of some & forEach
```suggestion
const index_check = (aValues, object) => {
  const TOKENIZER_ARRAY = aValues.flatMap(value => {
    if (value.includes('@index')) {
      if (value.slice(6, 7) !== '(' || value.slice(-1) !== ')') {
        throw new Error('@index is invalid, missing parenthesis or there are spaces in tokenizer.');
      }
      const FIELDS = value.slice(7, -1).split(',');
      return FIELDS.map(field =>field.trim());
    }
    return [];
  });
  if (TOKENIZER_ARRAY.length) {
    object.index = INDEX;
    object.tokenizer = TOKENIZER_ARRAY;
  }
};
```

Reply:
This won't work but I'll edit using map & flatmap

Reply:
Why this won't work ?

Reply:
INDEX is not defined and also eslint is not happy :) `length` property should be compared to a value

Reply:
If it's only this

:upside_down_face: 

```js
const index_check = (aValues, object) => {
  const TOKENIZER_ARRAY = aValues.flatMap(value => {
    if (value.includes('@index')) {
      if (value.slice(6, 7) !== '(' || value.slice(-1) !== ')') {
        throw new Error('@index is invalid, missing parenthesis or there are spaces in tokenizer.');
      }
      const FIELDS = value.slice(7, -1).split(',');
      return FIELDS.map(field =>field.trim());
    }
    return [];
  });
  if (TOKENIZER_ARRAY.length) {
    object.index = true;
    object.tokenizer = TOKENIZER_ARRAY;
  }
};
```